### PR TITLE
back-and-forth dcts for ac strategy

### PR DESCRIPTION
**Pull request by jyrkialakuijala**
measure the quality of an integral transform in the pixel space rather than dct coefficient base (as done before this PR)

some more ringing, better texture/subtle material fitting, some more details right, better performance on difficult images, cleaner architecture for further quality improvements, slower

```
Before:

Encoding      kPixels    Bytes          BPP  E MP/s  D MP/s     Max norm  SSIMULACRA2   PSNR        pnorm       BPP*pnorm   QABPP   Bugs
----------------------------------------------------------------------------------------------------------------------------------------
jxl:d0.1        20315 17381585    6.8445662   2.475  20.374   0.35752780  95.38110842  55.31   0.11907606  0.815023950090   6.845      0
jxl:d0.5        20315  6840469    2.6936578   3.043  30.431   0.81159277  91.28096152  46.43   0.35275973  0.950214025947   2.869      0
jxl:d0.8        20315  5026234    1.9792436   3.150  30.813   1.13390198  88.47341146  43.85   0.49912046  0.987880993221   2.364      0
jxl:d1          20315  4350486    1.7131458   3.204  32.057   1.34645549  86.76725072  42.66   0.58255077  0.997994400876   2.425      0
jxl:d1.5        20315  3367818    1.3261882   3.301  31.213   1.81159865  82.90450127  40.65   0.76549843  1.015194998864   2.587      0
jxl:d2.0        20315  2756702    1.0855414   3.351  31.893   2.25382353  79.17774155  39.19   0.94089680  1.021382378837   2.644      0
jxl:d2.5        20315  2341183    0.9219172   3.387  31.992   2.69914340  75.65279876  38.09   1.10692818  1.020496116869   2.695      0
jxl:d3          20315  2044812    0.8052114   3.169  31.749   3.11376459  72.28282385  37.21   1.26722218  1.020381792024   2.717      0
jxl:d5          20315  1363710    0.5370053   3.203  19.568   4.61445302  60.43067032  34.95   1.81935734  0.977004538421   2.621      0
jxl:d10         20315   803918    0.3165689   3.242  18.816   7.48235611  38.43593542  32.15   2.90685959  0.920221460214   2.439      0
Aggregate:      20315  3237961    1.2750529   3.142  27.277   1.87071131  74.92120226  40.60   0.76113960  0.970493289476   2.855      0

After

Encoding      kPixels    Bytes          BPP  E MP/s  D MP/s     Max norm  SSIMULACRA2   PSNR        pnorm       BPP*pnorm   QABPP   Bugs
----------------------------------------------------------------------------------------------------------------------------------------
jxl:d0.1        20315 17738232    6.9850076   1.837  20.854   0.36042851  95.39565073  55.62   0.11757951  0.821293744281   6.985      0
jxl:d0.5        20315  7083423    2.7893289   2.151  30.164   0.78012465  91.40832941  46.83   0.33259288  0.927710947397   2.885      0
jxl:d0.8        20315  5236376    2.0619939   2.235  30.836   1.12970444  88.63541402  44.21   0.47622603  0.981975169666   2.484      0
jxl:d1          20315  4502264    1.7729133   2.283  31.480   1.33640281  86.84913794  43.02   0.56322339  0.998546267198   2.511      0
jxl:d1.5        20315  3425292    1.3488205   2.347  30.762   1.87159566  82.76248986  40.94   0.77005819  1.038670260263   2.683      0
jxl:d2.0        20315  2780424    1.0948827   2.389  31.405   2.39023632  78.71547418  39.43   0.96306921  1.054447784649   2.773      0
jxl:d2.5        20315  2349487    0.9251872   2.434  31.658   2.84069056  75.09626979  38.26   1.13209238  1.047397331748   2.797      0
jxl:d3          20315  2029235    0.7990775   2.332  31.496   3.24060299  71.45589318  37.33   1.29583167  1.035469909914   2.738      0
jxl:d5          20315  1365069    0.5375405   2.358  19.596   4.49634149  59.74897514  35.13   1.81036984  0.973147025987   2.547      0
jxl:d10         20315   806188    0.3174628   2.352  18.654   7.78598920  36.65342191  32.20   2.97087931  0.943143743937   2.491      0
Aggregate:      20315  3288759    1.2950561   2.265  27.117   1.89966100  74.31700863  40.84   0.75648483  0.979690273899   2.922      0
```

### Pull Request Checklist

- [x] **CLA Signed**: Have you signed the [Contributor License Agreement](https://code.google.com/legal/individual-cla-v1.0.html) (individual or corporate, as appropriate)? Only contributions from signed contributors can be accepted.
- [x] **Authors**: If this is your first contribution, have you added your name to the [AUTHORS](AUTHORS) file?
- [x] **Code Style**: Have you ensured your code adheres to the project's coding style guidelines? You can use `./ci.sh lint` for automatic code formatting.

Please review the full [contributing guidelines](https://github.com/libjxl/libjxl/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: lib/jxl/enc_ac_strategy.cc [hunk 1]
```diff
@@ -468,27 +392,25 @@ float EstimateEntropy(const AcStrategy& acs, size_t x, size_t y,
         float qval = config.Quant(x / 8 + ix, y / 8 + iy);
         qval *= qval;
         qval *= qval;
+        qval *= qval;
         quant_norm8 += qval * qval;
```
Comment (szabadka) [inline lib/jxl/enc_ac_strategy.cc:396]: should be named quant_norm16 now

**Comment by mo271** [comment]
needs linting...

**Comment by mo271** [comment]
needs linting...

**Review by szabadka — APPROVED** [review]

File: lib/jxl/enc_ac_strategy.cc (hunk 1 above)
Comment (jyrkialakuijala) [reply lib/jxl/enc_ac_strategy.cc]: fixed

**Review by szabadka — APPROVED** [review]

**Pull request merged**